src/Identity/ApiAuthorization.IdentityServer/src/Extensions/DefaultClientRequestParametersProvider.cs

```diff
@@ -12,20 +12,24 @@ internal sealed class DefaultClientRequestParametersProvider : IClientRequestPar
 {
     public DefaultClientRequestParametersProvider(
         IAbsoluteUrlFactory urlFactory,
-        IOptions<ApiAuthorizationOptions> options)
+        IOptions<ApiAuthorizationOptions> options,
+        IIssuerNameService nameService)
     {
         UrlFactory = urlFactory;
         Options = options;
+        NameService = nameService;
     }
 
     public IAbsoluteUrlFactory UrlFactory { get; }
 
     public IOptions<ApiAuthorizationOptions> Options { get; }
 
+    public IIssuerNameService NameService { get; }
+
     public IDictionary<string, string> GetClientParameters(HttpContext context, string clientId)
     {
         var client = Options.Value.Clients[clientId];
-        var authority = context.GetIdentityServerIssuerUri();
+        var authority = NameService.GetCurrentAsync().GetAwaiter().GetResult();
```

Comment:
This is a little gross but the old obsolete API is sync, and IIssuerNameService is async only

Reply:
At what point can we make breaking changes and make this one async? Isn't a version like 7.0 a good opportunity? Or at least add/use the async one and make the `GetClientParameters` obsolete.